src/Morphic-Core/Morph.class.st

```diff
@@ -5743,6 +5785,95 @@ Morph >> simulateKeyStrokes: aString [
 	aString do: [:c | self simulateKeyStroke: c ].
 ]
 
+{ #category : #'event testing' }
+Morph >> simulateMiddleClick [
+
+	self simulateClickWith: MouseEvent blueButton.
+]
+
+{ #category : #'event testing' }
+Morph >> simulateMiddleMouseDown [ 
+	"Simulate mouse button down with the left button (red button for left, blue for middle, yellow for right)"
+	self simulateMouseDownWith: MouseEvent blueButton .
+]
+
+{ #category : #'event testing' }
+Morph >> simulateMiddleMouseUp [ 
+	"Simulate mouse button up with the left button (red button for left, blue for middle, yellow for right)"
+	self simulateMouseUpWith: MouseEvent blueButton .
+]
+
+{ #category : #'event testing' }
+Morph >> simulateMouseDown [ 
+	"Simulate mouse button down with the left button (red button for left, blue for middle, yellow for right)"
+	self simulateMouseDownWith: MouseEvent redButton.
+]
+
+{ #category : #'event testing' }
+Morph >> simulateMouseDownWith: buttons [
+	"Simulate mouse button down with a given button"
+	
+		self activeHand handleEvent: (MouseButtonEvent new 
+			setType: #mouseDown
+			position: (self pointInWorld: self center) "Some Morphs report local coords"
+			which:  buttons
+			buttons: buttons
+			hand: self activeHand
+			stamp: Time millisecondClockValue)
+]
+
+{ #category : #'event testing' }
+Morph >> simulateMouseMove: aPointStart to: aPointEnd [
+"Simulates a mouse moving from a given point to a destinationpoint"
+	| event |
+	event := MouseMoveEvent  new 
+				setType: #mouseMove  
+				startPoint: aPointStart 
+				endPoint: aPointEnd 
+				trail: nil
+				buttons: 0
+				hand: ActiveHand 
```

Comment:
this one should be `self activeHand`

Reply:
Hummm, is curious, that runs on my image, and the existing "simulateKeyStroke" also uses ActiveHand instead of self activeHand.

I will try to change both, but don't undertand why I don't get an error on my image.

Reply:
I changed that this week... sorry for the noise.

Reply:
ok.